contracts/transfer/circuits/src/execute.rs

```diff
@@ -0,0 +1,359 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::gadgets;
+
+use crossover::CircuitCrossover;
+use input::{CircuitInput, WitnessInput};
+use output::{CircuitOutput, WitnessOutput};
+
+use anyhow::Result;
+use canonical::Store;
+use dusk_plonk::bls12_381::BlsScalar;
+use dusk_plonk::jubjub::JubJubExtended;
+use phoenix_core::Note;
+use poseidon252::sponge::sponge::sponge_hash_gadget;
+use poseidon252::tree::{
+    self, PoseidonLeaf, PoseidonTree, PoseidonTreeAnnotation,
+};
+use rand_core::{CryptoRng, RngCore};
+use schnorr::gadgets as schnorr_gadgets;
+
+use dusk_plonk::prelude::*;
+
+mod crossover;
+mod input;
+mod output;
+
+#[cfg(test)]
+mod tests;
+
+/// The circuit responsible for creating a zero-knowledge proof
+/// for a 'send to contract transparent' transaction.
+#[derive(Debug, Clone)]
+pub struct ExecuteCircuit<const DEPTH: usize> {
```

Comment:
I belive @Bounce23 has the numbers to do the following:

Instead of requesting the `trim_size` to create the circuit with capacity, ask for the fields and matching on the len, set the propper trim_size.

Reply:
This is correct, they can be run on shorter sizes. I'll update the numbers in Gitbook, as they are pre schnorr sizes. 
I would also like to check if there any more changes that will affect the size.

Reply:
I set the tests with the minimum possible size for the circuit. I'll update the scripts to generate the circuits, and this is the next step